### ISSUE TITLE
Report deprecated rules that are in use

### ISSUE THREAD

**Issue by randycoulman** [sarbbottam/eslint-find-rules#188]
In addition to #172, which would keep eslint-find-rules from reporting errors when deprecated rules are unused, it might also be nice to have a command-line option that shows deprecated rules that are still being used in a configuration.

**Comment by randycoulman** [comment]
I've been thinking more about this as I think about implementing it, and I realize that eslint already warns when deprecated rules are in use.  Is this feature useful enough to implement here anyway?  Or should I just close the issue?

**Comment by ljharb** [comment]
I think it's still useful to get a zero/nonzero exit code based on the config files, without having to actually run eslint.

**Comment by randycoulman** [comment]
Yeah, good point.  I'll keep it open and work on it when I get a chance.

**Comment by alexilyaev** [comment]
ESLint doesn't warn when using deprecated rules in the config, only when using rules that don't exist anymore.

I've got 2 deprecated rules here: https://github.com/alexilyaev/react-es6-starter
`newline-after-var` and `newline-before-return`
Running `eslint -c .eslintrc --ignore-path .gitignore .` doesn't show me any warning.

I think an option to show currently configured deprecated rules is very useful.  
We actually have no way of checking it right now

I'm doing something similar with [stylelint-find-rules](https://github.com/alexilyaev/stylelint-find-rules).  
Can show configured deprecated rules.

Just tried filtering the `currentRules`:
```
  this.getCurrentDeprecatedRules = function () {
    var rules = eslint.linter.getRules();

    return getSortedRules(currentRules).filter((rule) => {
      return _isDeprecated(rules.get(rule) || '')
    });
  };
```

It worked, though if I have extends, it counts rules from them as well.  
Perhaps ignore extends somehow.

**Comment by ta2edchimp** [comment]
A flag to ignore a config's "extends" would be nice.  
But atm, only manipulating the input (removing the "extends" property of the config to be loaded) before actually loading the configuration via ESLint's CLIEngine comes to my mind.

**Comment by alexilyaev** [comment]
@ta2edchimp @randycoulman So, after 2 years, I find myself back here :-)

I'm trying to validate user configured rules to not have deprecated rules.  
This is useful when upgrading configs/plugins which deprecate certain rules.

Right now running `eslint-find-rules -d .eslintrc` will show deprecated rules not only from user configured rules, but also from `extends`.

I dug into the code and found it's not so simple to mitigate.  
At `_getConfigs` we use `cliEngine.getConfigForFile` which loads the `configFile` and merges all `extends` on top of it.  
Which means that `currentRuleNames` will already include rule from extends and not only the user defined rules.  
And then `deprecatedRuleNames` can show rules that were not defined by the user.

This happened to me with `eslint-config-prettier`:
https://github.com/prettier/eslint-config-prettier/issues/112

I tried to dig into `eslint`'s `CLIEngine` code as well.  
Looks like there's a cascade of configs that are being loaded and merged under `CascadingConfigArrayFactory`.  
But eventually our `configFile` is loaded with `configArrayFactory.loadFile`, which runs `_normalizeConfigData`, which then loads and merges the `extends` and other things.

At this point, I'm stuck and need some help.
